src/promptflow-tracing/promptflow/tracing/_trace.py

```diff
@@ -402,13 +376,19 @@ async def wrapped(*args, **kwargs):
                 Tracer.push(trace)
                 enrich_span_with_input(span, trace.inputs)
                 output = await func(*args, **kwargs)
-                output = enrich_span_with_trace_type(span, trace.inputs, output, trace_type)
-                span.set_status(StatusCode.OK)
-                output = Tracer.pop(output)
+                if isinstance(output, AsyncIterator):
+                    output = Tracer.pop(output)
+                    output = traced_async_generator(span, trace.inputs, output, trace_type)
+                else:
+                    enrich_span_with_trace_type(span, trace.inputs, output, trace_type)
+                    span.set_status(StatusCode.OK)
+                    output = Tracer.pop(output)
             except Exception as e:
                 Tracer.pop(None, e)
                 raise
         token_collector.collect_openai_tokens_for_parent_span(span)
+        if not isinstance(output, AsyncIterator):
```

Comment:
We should end the span even an exception is raised?

Reply:
Try ... finally?

Reply:
We plan to leverage the `start_as_current_span` method, which already handle the exception handling in the context manger to achieve this goal. Instead, we will implement our own start_as_current_span context manager that wraps logic of span lifecycle management.